app/controllers/ems_common.rb

```diff
@@ -311,16 +317,25 @@ def update_button_save
               {:model => ui_lookup(:model => model.to_s), :name => update_ems.name}
       AuditEvent.success(build_saved_audit(update_ems, @edit))
       session[:edit] = nil  # clean out the saved info
-      javascript_redirect :action => 'show', :id => @ems.id.to_s, :flash_msg => flash
+      if restful_routed?(model)
+        javascript_redirect polymorphic_path(model.find(params[:id]), :flash_msg => flash)
+      else
+        javascript_redirect :action => 'show', :id => @ems.id.to_s, :flash_msg => flash
+      end
       return
     else
       @edit[:errors].each { |msg| add_flash(msg, :error) }
       update_ems.errors.each do |field, msg|
         add_flash("#{field.to_s.capitalize} #{msg}", :error)
       end
+
+      breadcrumb_url = "/#{@table_name}/edit/#{@ems.id}"
```

Comment:
those 2 lines seem like more suitable with if/else rather than doing a double assignment to breadcrumb_url